third_party/CMakeLists.txt

```diff
@@ -58,10 +58,22 @@ if(VELOX_ENABLE_ARROW)
                                       ${THRIFT_INCLUDE_DIR})
   set_property(TARGET thrift PROPERTY IMPORTED_LOCATION ${THRIFT_LIB})
 
+  if(DEFINED ENV{VELOX_ARROW_URL})
+    set(ARROW_SOURCE_URL "$ENV{VELOX_ARROW_URL}")
+  else()
+    set(VELOX_ARROW_BUILD_VERSION 8.0.0)
+    set(ARROW_SOURCE_URL
+        "https://dlcdn.apache.org/arrow/arrow-${VELOX_ARROW_BUILD_VERSION}/apache-arrow-${VELOX_ARROW_BUILD_VERSION}.tar.gz"
+    )
+  endif()
+  set(VELOX_ARROW_BUILD_SHA256_CHECKSUM
+      ad9a05705117c989c116bae9ac70492fe015050e1b80fb0e38fde4b5d863aaa3)
+
   ExternalProject_Add(
     arrow_ep
     PREFIX ${ARROW_PREFIX}
```

Comment:
I think if the env property VELOX_ARROW_URL is set then we should also have the env property VELOX_ARROW_*_CHECKSUM set or not check for checksum if an external URL is provided and assume that the dev knows what he is doing. 

cc: @assignUser

Reply:
Great!

When `${dependency}_SOURCE` is `BUNDLED`, we have 2 choses:
1. let user set env property VELOX_PACKAGE_URL and VELOX_PACKAGE_CHECKSUM.
2. Velox keep specific version, provide a global dependency version and checksum, just like [Arrow versions.txt](https://github.com/apache/arrow/blob/main/cpp/thirdparty/versions.txt). User just need pass correct VELOX_PACKAGE_URL.

I prefer chose 2, what community's suggestion?

Reply:
I agree with @kgpai: set the checksum from envvar if set otherwise skip version verification.

The easiest way to do this on first glance seems to be to pull `URL_HASH SHA256=` into the checksum variable and just skip setting it if the envvar does not exists so the whole option is skipped.

Something like:
```
if(DEFINED ENV{VELOX_ARROW_SHA256})
  set(VELOX_ARROW_BUILD_CHECKSUM "URL_HASH SHA256=$ENV{VELOX_ARROW_SHA256}")
endif()
``` (I named the envvar sha256 to make it obvious which checksum to use)